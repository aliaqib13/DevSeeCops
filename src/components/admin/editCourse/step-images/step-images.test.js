import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import StepImages from '.';

const props = {
    fetchStepsImages: jest.fn(() => Promise.resolve(true)),
    images: [
        {
            course_id: 1,
            id: 4,
            image: 'https://atp-resources-dev-testing.s3.eu-central-1.amazonaws.com/steps-images/EIaG35P0jcq1VTEX9UdrUGAEIUKFtOve.png',
            uuid: '6c2a6c73-da92-441d-a9ce-369b09d28fa5',
        },
        {
            course_id: 1,
            id: 5,
            image: 'https://atp-resources-dev-testing.s3.eu-central-1.amazonaws.com/steps-images/tvSyAhFtu5QWvqpbZLxs9Y4IYNOctsW9.png',
            uuid: '6487959d-6fbf-4c3a-811c-756950047c39',
        },
        {
            course_id: 1,
            id: 6,
            image: 'https://atp-resources-dev-testing.s3.eu-central-1.amazonaws.com/steps-images/yOUCX4WNi9oRM5Nmm6YSayoK3NZafg9h.png',
            uuid: '644d54b1-a654-4a28-b975-bb191122c233',
        },
    ],
};

describe('Step images', () => {
    let component;

    beforeEach(() => {
        component = shallow(<StepImages {...props} />);
    });

    it('Should render successfully', () => {
        const wrapper = component.find('.step-images');
        expect(wrapper.exists()).toBeTruthy();
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Should render images passed down by props', () => {
        const list = component.find('List');
        expect(list.props().dataSource).toBe(props.images);
    });
});

describe('Steps images list', () => {
    let component;

    beforeEach(() => {
        component = mount(<StepImages {...props} />);
    });

    it('Should render list with proper images and descriptions', () => {
        const list = component.find('List');
        const listItem = list.find('Item').at(1);
        expect(listItem.find('Avatar').props().src).toBe(props.images[1].image);
        expect(listItem.find('.ant-list-item-meta-description').text()).toBe(props.images[1].image);
    });

    it('Should open preview modal with proper image when clicked on preview icon of image item in the list', () => {
        const list = component.find('List');
        const previewIcon = list.find('Button.img-preview-button').at(0);
        expect(component.find('Modal.steps-image-preview-modal').props().visible).toBe(false);
        previewIcon.simulate('click');
        expect(component.find('Modal.steps-image-preview-modal').props().visible).toBeTruthy();
        expect(component.find('Modal.steps-image-preview-modal').find('img').props().src).toBe(props.images[0].image);
    });

    it('Should call removeStepImage method when click on delete icon of image item in the list', () => {
        const list = component.find('List');
        const deleteIcon = list.find('Button.img-delete-button').at(0);
        const spy = jest.spyOn(component.instance(), 'removeStepImage');
        deleteIcon.simulate('click');
        expect(spy).toBeCalledTimes(1);
    });
});
