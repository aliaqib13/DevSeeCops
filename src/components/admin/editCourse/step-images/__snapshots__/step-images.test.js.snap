// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step images Should render successfully 1`] = `
<div
  className="step-images"
>
  <UploadImage
    uploadImage={[Function]}
  />
  <List
    bordered={false}
    className="steps-images-list"
    dataSource={
      Array [
        Object {
          "course_id": 1,
          "id": 4,
          "image": "https://atp-resources-dev-testing.s3.eu-central-1.amazonaws.com/steps-images/EIaG35P0jcq1VTEX9UdrUGAEIUKFtOve.png",
          "uuid": "6c2a6c73-da92-441d-a9ce-369b09d28fa5",
        },
        Object {
          "course_id": 1,
          "id": 5,
          "image": "https://atp-resources-dev-testing.s3.eu-central-1.amazonaws.com/steps-images/tvSyAhFtu5QWvqpbZLxs9Y4IYNOctsW9.png",
          "uuid": "6487959d-6fbf-4c3a-811c-756950047c39",
        },
        Object {
          "course_id": 1,
          "id": 6,
          "image": "https://atp-resources-dev-testing.s3.eu-central-1.amazonaws.com/steps-images/yOUCX4WNi9oRM5Nmm6YSayoK3NZafg9h.png",
          "uuid": "644d54b1-a654-4a28-b975-bb191122c233",
        },
      ]
    }
    itemLayout="horizontal"
    loading={false}
    pagination={true}
    renderItem={[Function]}
    split={true}
  />
  <Modal
    className="steps-image-preview-modal"
    confirmLoading={false}
    footer={
      Array [
        <Button
          block={false}
          ghost={false}
          htmlType="button"
          loading={false}
          onClick={[Function]}
        >
          Close
        </Button>,
      ]
    }
    maskTransitionName="fade"
    okType="primary"
    onCancel={[Function]}
    title="Image Preview"
    transitionName="zoom"
    visible={false}
    width="50%"
  >
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <img
        alt="step"
        src=""
        style={
          Object {
            "maxWidth": "100%",
          }
        }
      />
    </div>
  </Modal>
</div>
`;
